include/deal.II/distributed/shared_tria.h

```diff
@@ -460,6 +460,71 @@ namespace parallel
 #endif
 } // namespace parallel
 
+
+namespace internal
+{
+  namespace parallel
+  {
+    namespace shared
+    {
+      /**
+       * This class temporarily modifies the subdomain ID of all active cells to
+       * their respective "true" owner.
+       *
+       * The modification only happens on parallel::shared::Triangulation
+       * objects with artificial cells, and persists for the lifetime of an
+       * instantiation of this class.
+       *
+       * The SubdomainModifier class should only be used for temporary read-only
+       * purposes. For example, whenever your implementation requires to treat
+       * artificial cells temporarily as locally relevant to access their dof
+       * indices.
+       *
+       * This class will not do anything when artificial cells are allowed.
```

Comment:
Did you mean "are not allowed"?

Reply:
Yes, you are right. But I just realized this forms a double negation. I'll go with something like `will only operate when they are allowed`.